benchmark/utils/stencil_matrix.hpp

```diff
@@ -209,11 +210,11 @@ gko::matrix_data<ValueType, IndexType> generate_3d_stencil_box(
         gko::dim<2>{static_cast<gko::size_type>(global_size),
                     static_cast<gko::size_type>(global_size)});
 
-    auto global_offset = [&](const int position_x, const int position_y,
-                             const int position_z) {
-        return position_x * static_cast<int>(local_size) * dims[0] * dims[1] +
+    auto global_offset = [&](const int position_z, const int position_y,
+                             const int position_x) {
+        return position_x * static_cast<int>(local_size) +
                position_y * static_cast<int>(local_size) * dims[0] +
-               position_z * static_cast<int>(local_size);
+               position_z * static_cast<int>(local_size) * dims[0] * dims[1];
```

Comment:
global_offset may overflow?

Reply:
Thank you, good point. I will cast everything to `IndexType`, that should fix it.